### ISSUE TITLE
IntermediateGroup returns a group but IntermediateSubgroups returns empty list

### ISSUE THREAD

**Issue by hwatheod** [gap-system/gap#4851]
### Observed behaviour
```
gap> M24 := MathieuGroup(24);;
gap> s2 := SylowSubgroup(M24, 2);;
gap> v2 := IntermediateGroup(M24, s2); # takes some time to run
<permutation group with 7 generators>
gap> w2 := IntermediateSubgroups(M24, s2);
rec( inclusions := [  ], subgroups := [  ] )
```
### Expected behaviour

`IntermediateSubgroups(M24, s2)` should find at least the group found by `IntermediateGroup(M24, s2)`.

This looks similar to the issues #4246 and #4395 but I'm not sure if it's the same. In the case of the example in  #4395, the IntermediateGroup command `IntermediateGroup(g, hs[1])` returns `fail` whereas in this example it returns an actual group.

### Copy and paste GAP banner (to tell us about your setup)
```
 ┌───────┐   GAP 4.11.1 of 2021-03-02
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin18.7.0-default64-kv7
 Configuration:  gmp 6.2.1, GASMAN
 Loading the library and packages ...

 Packages:   AClib 1.3.2, Alnuth 3.1.2, AtlasRep 2.1.0, AutoDoc 2020.08.11, 
             AutPGrp 1.10.2, Browse 1.8.11, CaratInterface 2.3.3, CRISP 1.4.5, 
             Cryst 4.1.23, CrystCat 1.1.9, CTblLib 1.3.1, FactInt 1.6.3, 
             FGA 1.4.0, Forms 1.2.5, GAPDoc 1.6.4, genss 1.6.6, IO 4.7.0, 
             IRREDSOL 1.4.1, LAGUNA 3.9.3, orb 4.8.3, Polenta 1.3.9, 
             Polycyclic 2.16, PrimGrp 3.4.1, RadiRoot 2.8, recog 1.3.2, 
             ResClasses 4.7.2, SmallGrp 1.4.2, Sophus 1.24, SpinSym 1.5.2, 
             TomLib 1.2.9, TransGrp 3.0, utils 0.69, XGAP 4.30
```

**Comment by fingolfin** [comment]
This is fixed on `master`, which will hopefully "soon" be released as GAP 4.12.0. It is also much faster (~10 milliseconds on my computer)
```
gap> M24 := MathieuGroup(24);;
gap> s2 := SylowSubgroup(M24, 2);;
gap> v2 := IntermediateGroup(M24, s2);
<permutation group of size 9216 with 12 generators>
gap> time;
10
gap> w2 := IntermediateSubgroups(M24, s2);
rec( inclusions := [ [ 0, 1 ], [ 0, 2 ], [ 0, 3 ], [ 0, 4 ], [ 1, 5 ], [ 1, 6 ],
      [ 1, 8 ], [ 2, 5 ], [ 2, 9 ], [ 3, 7 ], [ 3, 8 ], [ 3, 9 ], [ 4, 6 ], [ 4, 7 ],
      [ 5, 11 ], [ 5, 12 ], [ 6, 10 ], [ 6, 11 ], [ 7, 10 ], [ 8, 10 ], [ 8, 12 ],
      [ 9, 12 ], [ 10, 13 ], [ 11, 13 ], [ 12, 13 ] ],
  subgroups := [ <permutation group of size 3072 with 11 generators>,
      <permutation group of size 3072 with 11 generators>,
      <permutation group of size 3072 with 5 generators>,
      <permutation group of size 3072 with 11 generators>,
      <permutation group of size 9216 with 7 generators>,
      <permutation group of size 9216 with 6 generators>,
      <permutation group of size 9216 with 6 generators>,
      <permutation group of size 21504 with 4 generators>,
      <permutation group of size 21504 with 4 generators>,
      Group([ (1,9)(2,6)(3,23)(4,19)(5,7)(8,18)(10,15)(11,22)(12,24)(13,17)(14,21)
          (16,20), (1,13,20,12)(2,8,10,16)(3,22,6,4)(5,18,19,11)(7,21)(9,23)(14,17)
          (15,24) ]),
      Group([ (1,4)(2,15)(3,21)(5,16)(6,23)(7,12)(8,19)(9,13)(10,14)(11,24)(17,18)
          (20,22), (1,10,17,18,23,24,11,15,16,5,3,4)(2,8,21,22,14,7)(6,9)(12,19,13,20) ])
        ,
      Group([ (1,5,13)(2,17,16)(3,7,22)(4,14,6)(8,15,10)(12,20,18), (1,19,12,13,11,20)
          (2,3,14,17,24,4,10,6,23,9,7,22)(5,18)(8,21,16,15) ]) ] )
gap> time;
348
```

**Comment by hulpke** [comment]
Since this is already fixed in master, I close
